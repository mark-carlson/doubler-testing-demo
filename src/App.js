import React, { useState } from 'react';
import { doubleNumber } from './helperFunctions'
import './App.css';

function App() {
  const [numberDisplay, setNumberDisplay] = useState('1');

  return (
    <div className="App jumbotron">
      <h1>React Doubler</h1>
      <h2 className="numberDisplay" data-testid="numberDisplay">{numberDisplay}</h2>
      <div className="btn-container">
        <button id="doubler-btn" className="btn btn-primary" onClick={() => setNumberDisplay(doubleNumber(numberDisplay))}>
          Doubler
        </button>
      </div>
    </div>
  );
}

export default App;
