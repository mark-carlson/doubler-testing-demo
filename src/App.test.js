import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {
  render,
  fireEvent
} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import App from './App';

describe('App', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should accept a click on the button', () => {
    const { getByText, container, getByTestId } = render(<App/>);
    expect(container).toMatchSnapshot();
    fireEvent.click(getByText('Doubler'));
    expect(getByTestId('numberDisplay')).toHaveTextContent('2');
    fireEvent.click(getByText('Doubler'));
    expect(getByTestId('numberDisplay')).toHaveTextContent('4');
  })
})
