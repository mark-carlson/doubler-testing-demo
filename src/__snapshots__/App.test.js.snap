// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should accept a click on the button 1`] = `
<div>
  <div
    class="App jumbotron"
  >
    <h1>
      React Doubler
    </h1>
    <h2
      class="numberDisplay"
      data-testid="numberDisplay"
    >
      1
    </h2>
    <div
      class="btn-container"
    >
      <button
        class="btn btn-primary"
        id="doubler-btn"
      >
        Doubler
      </button>
    </div>
  </div>
</div>
`;

exports[`App should render correctly 1`] = `
<div
  className="App jumbotron"
>
  <h1>
    React Doubler
  </h1>
  <h2
    className="numberDisplay"
    data-testid="numberDisplay"
  >
    1
  </h2>
  <div
    className="btn-container"
  >
    <button
      className="btn btn-primary"
      id="doubler-btn"
      onClick={[Function]}
    >
      Doubler
    </button>
  </div>
</div>
`;
